### ISSUE TITLE
Books shop items sorting

### ISSUE THREAD

**Issue by igroglaz** [draconisPW/PWMAngband#23]
https://cdn.discordapp.com/attachments/535396965433868288/640236237399130112/Untitled-1.jpg

In custom shop with books dialogs are in bottom part, although in store.txt they are on top

```
# Starter town  old wizard 
store:Rinswind the Wizard
owner:5000:Rinswind the Wizard (Human)
slots:0:3
turnover:9
always:food:Who are ya? What do ya need? Whaat? I can't hear you..
always:food:*grumbles* They come.. and go.. come.. and sometimes not come back..
always:food:*whipped up* So.. Can ya read? If you can't, I can't help ya..
always:food:But if you do - take a look at this finest books about miracles.
always:food:With them you would be able to conquer world!.. What?? 
always:food:I can't hear ya!! Speak louder, yongster! Whaat?
always:nature book
always:shadow book
always:psi book
always:elemental book
buy:nature book
buy:shadow book
buy:psi book
buy:elemental book
...
```

**Comment by draconisPW** [comment]
Items are sorted by descending tval, which means that they appear in the reverse order from object_base.txt. Since the tvals are hardcoded in list-tvals.h, there is nothing that can be done.

I will provide a workaround to this by adding a new feature: fortune cookies. In regular PWMangband, a fortune cookie while consumed will give a random hint. I'll put the tval at the end so using this will alwayd be at the beginning of any store inventory.

**Comment by draconisPW** [comment]
Implemented in 746e7ed.
